src/App.tsx

```diff
@@ -1,24 +1,208 @@
+import { ErrorNotification } from
+  './components/ErrorNotification/ErrorNotification';
+import { ErrorType } from './types/ErrorType';
 
-const USER_ID = 0;
+export const USER_ID = 6344;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [numberOfActive, setNumberOfActive] = useState(0);
+  const [numberOfCompleted, setNumberOfCompleted] = useState(0);
+  const [errorType, setErrorType] = useState<ErrorType>(ErrorType.NONE);
+  const [hasError, setHasError] = useState(false);
+  const [filterValue, setFilterValue] = useState<Filter>(Filter.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processingTodosIds, setProcessingTodosIds] = useState<number[]>([]);
+
+  const showError = useCallback((message: ErrorType) => {
+    setErrorType(message);
+    setHasError(true);
+    setTimeout(() => {
+      setHasError(false);
+    }, 3000);
+  }, []);
+
+  const fetchTodos = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      showError(ErrorType.LOAD);
+    }
+  }, []);
+
+  useEffect(() => {
+    fetchTodos();
+  }, []);
+
+  useEffect(() => {
+    const counActive = todos.filter(todo => !todo.completed).length;
+    const countCompleted = todos.length - counActive;
+
+    setNumberOfActive(counActive);
+    setNumberOfCompleted(countCompleted);
+  }, [todos]);
+
+  const visibleTodos = useMemo(() => (
+    getVisibleTodos(filterValue, todos)
+  ), [filterValue, todos]);
+
+  const handleFilterSelection = useCallback((value:Filter) => {
+    setFilterValue(value);
+  }, []);
+
+  const handleClearCompleted = async () => {
+    try {
+      await Promise.all(
+        todos.filter(todo => todo.completed).map(todo => {
+          setProcessingTodosIds(currentIds => [...currentIds, todo.id]);
+
+          return deleteTodo(todo.id);
+        }),
+      );
+    } catch {
+      showError(ErrorType.DELETE);
+    }
+
+    await fetchTodos();
+    setProcessingTodosIds([]);
+  };
+
+  const handleMultipleToogle = async () => {
+    try {
+      await Promise.all(
+        todos
+          .filter(todo => {
+            return (numberOfActive) ? !todo.completed : todo.completed;
+          })
+          .map(({ id, completed }) => {
+            setProcessingTodosIds((currentIds) => [...currentIds, id]);
+
+            return patchTodo(id, { completed: !completed });
+          }),
+      );
+    } catch {
+      showError(ErrorType.UPDATE);
+    }
+
+    await fetchTodos();
+    setProcessingTodosIds([]);
+  };
+
+  const handleTodoDeletion = useCallback(async (todoId: number) => {
+    setProcessingTodosIds((currentIds) => [...currentIds, todoId]);
+
+    try {
+      await deleteTodo(todoId);
+    } catch {
+      showError(ErrorType.DELETE);
+    }
+
+    await fetchTodos();
+    setProcessingTodosIds([]);
```

Comment:
only remove what you added, dont reset the array

Reply:
here what it leads to
![ToDo App](https://user-images.githubusercontent.com/117474961/221158769-7f3ddcb3-bd4e-41c4-9ead-50cc39e0500d.gif)